src/server/pfs/server/driver_file.go

```diff
@@ -129,6 +133,12 @@ func (d *driver) openCommit(ctx context.Context, commit *pfs.Commit, opts ...ind
 	if err != nil {
 		return nil, nil, err
 	}
+	if commitInfo.Finishing != nil && commitInfo.Finished == nil {
+		_, err := d.inspectCommit(ctx, commit, pfs.CommitState_FINISHED)
```

Comment:
So this is going to block, while the parent compacts.  I guess that's okay.

Reply:
Yeah, the alternative is what we have now which is that any call that is made before a separate call compacts will duplicate the work. This seems like the ideal model.